core/src/components/toggle/toggle.tsx

```diff
@@ -285,6 +290,33 @@ export class Toggle implements ComponentInterface {
     );
   }
 
+  /**
+   * On Safari (iOS 18+) we can trigger haptic feedback programatically
+   * by rendering <input type="checkbox" switch> element
+   * with an associated <label>  and triggering click()
+   * on the <label> element.
+   */
+  private renderFallbackHapticElements() {
+    const { inputId } = this;
+    const mode = getIonMode(this);
+
+    if (hapticAvailable() || mode !== 'ios') {
```

Comment:
If I understand correctly, we don't need these elements if Vibration API is available, because then `hapticSelection()` will work.